server/src/main/java/org/candlepin/sync/RulesImporter.java

```diff
@@ -53,6 +56,9 @@ public void importObject(Reader reader) throws IOException {
                 existingRules.getVersion() + " new version: " + newRules.getVersion());
             curator.update(newRules);
             sink.emitRulesModified(existingRules, newRules);
+
+            // Trigger a recompile of the rules:
+            jsProvider.get();
```

Comment:
Could we just make compileRules public and call that directly? .get appears to create some context and a new runner. If we're only interested in a recompile, we could save ourselves a bit of work there.